atlasdb-ete-tests/build.gradle

```
@@ -94,50 +94,111 @@ dependencies {
     testCompileOnly 'org.immutables:value::annotations'
 }
 
+// ETE Test Wiring
+//
+// For each non-trivially sized ETE test and all ETE test suites:
+// - define a custom gradle task for just that test [suite]
+// - remove that test [suite] from :atlasdb-ete-tests:test
+// - add that custom task as a dependency of :customEteTasks, which :atlasdb-ete-tests:test then depends on
+// - use task exclusion in our circleci commands to ensure only specific things get run in certain containers at CI time
+//
+// This does a few things:
+// - for test suites, makes it easier to reason about which suite class executions belong to what test suites in the
+//   generated junit reports and test suite cost in gradle build scans
+// - allows us to run certain tests in their own containers for CI speed reasons
+// - on the off chance someone adds or re-names a test in the future, but forgets to give it its own task and wiring
+//   described above, that test will still get run via :atlasdb-ete-tests:test to avoid a situation where the new test
+//   isn't run in CI without us realizing
+
+// Tasks required before ETE tests can be run
+
 task prepareForEteTests(type: Copy, dependsOn: 'distTar') {
     from distTar.outputs
     into 'build/docker/'
 
     rename { filename -> 'atlasdb-ete-snapshot.tgz' }
 }
 
-task longTest(type: Test) {
-    dependsOn prepareForEteTests
-    include '**/MultiCassandraTestSuite.class'
+task prepareForTimelockEteTests() {
+    dependsOn 'prepareForEteTests', ':timelock-server-distribution:dockerTag'
 }
 
-task timeLockTest(type: Test) {
-    dependsOn prepareForEteTests, ':timelock-server-distribution:dockerTag'
-    include '**/*TimeLock*.class'
-    exclude '**/*TimeLockMigration*.class'
+// Individual ETE Test Tasks
+
+task timeLockMigrationTest(type: Test, dependsOn: 'prepareForTimelockEteTests') {
```

I generally find it preferable to keep all task configuration in the closure.
```suggestion
task timeLockMigrationTest(type: Test) {
    dependsOn 'prepareForTimelockEteTests
```

This is the correct way to configure tasks using [non-deprecated and recommended API](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:defining_tasks).
```suggestion
tasks.register('timeLockMigrationTest', Test) {
    dependsOn 'prepareForTimelockEteTests
```